src/legacy/core_plugins/kibana/public/discover/controllers/discover.js

```diff
@@ -91,7 +91,7 @@ const app = uiModules.get('apps/discover', [
 
 uiRoutes
   .defaults(/^\/discover(\/|$)/, {
-    requireDefaultIndex: true,
```

Comment:
When setting this to false, kibana no longer is trying to persist the `defaultIndex` when none is set 

https://github.com/elastic/kibana/blob/fcc4c4878f51e18177b59e2294c7c662967073a1/src/legacy/core_plugins/kibana/public/management/route_setup/load_default.js#L89